packages/browser/scripts/checkbrowsers.js

```diff
@@ -0,0 +1,132 @@
+    }
+  });
+};
+
+const isMatchingEntry = (key, localConfig, bsConfig) => {
+  let localValue = localConfig[key];
+  let bsValue = bsConfig[key];
+
+  // all values are either null, undefined, or strings, so this should only catch strings
+  if (localValue) {
+    localValue = localValue.toLowerCase();
+  }
+  if (bsValue) {
+    bsValue = bsValue.toLowerCase();
+  }
+
+  if (localValue === bsValue) {
+    return true;
+  }
+  if (key === 'browser_version' && localValue === 'latest') {
+    return true;
+  }
+
+  return false;
+};
+
+const isMatchingConfig = (localConfig, bsConfig) => {
+  const checkKeys = ['os', 'os_version', 'browser', 'device', 'browser_version'];
+
+  for (const key of checkKeys) {
+    if (!isMatchingEntry(key, localConfig, bsConfig)) {
+      return false;
+    }
+  }
+
+  // while we're here, if we've found a match on everything else, make sure
+  // real_mobile is up to date. Now the data *really* matches!
+  if (localConfig.real_mobile !== bsConfig.real_mobile) {
+    localConfig.real_mobile_updated = true; // flag for later
+    localConfig.real_mobile = bsConfig.real_mobile;
+  }
+
+  return true;
+};
+
+const isSupported = (localConfig, supportedConfigs) => {
+  return supportedConfigs.some(supportedConfig => isMatchingConfig(localConfig, supportedConfig));
+};
+
+const checkLocalConfigsVsBrowserStack = (localConfigs, bsConfigs) => {
+  const unsupportedConfigs = [];
+  const realMobileUpdates = [];
+
+  // check each local config against the entire collection of BS configs
+  for (const configName in localConfigs) {
+    const localConfig = localConfigs[configName];
+
+    console.log(`\nChecking ${configName}`);
+
+    if (!isSupported(localConfig, bsConfigs)) {
+      console.log('  UNSUPPORTED');
+      unsupportedConfigs.push(configName);
+    } else if (localConfig.real_mobile_updated) {
+      console.log('  Supported (but needs real_mobile update)');
+      realMobileUpdates.push(configName);
+    } else {
+      console.log('  Supported!');
+    }
+  }
+
+  // report on unsupported configs
+  if (unsupportedConfigs.length) {
+    console.log('\nFound unsupported browser configurations:');
+    for (const configName of unsupportedConfigs) {
+      console.log(`\n${configName}: `, localConfigs[configName]);
+    }
+    console.log(
+      '\nPlease visit https://api.browserstack.com/5/browsers or https://api.browserstack.com/5/browsers?flat=true to choose new configurations.',
+    );
+  } else {
+    console.log('\nAll configurations supported!\n');
+  }
+
+  // report on real_mobile updates
+  if (realMobileUpdates.length) {
+    console.log('\nFound supported browser configurations which need real_mobile updated:\n');
+    for (const configName of realMobileUpdates) {
+      console.log(configName, 'new real_mobile value: ', localConfigs[configName].real_mobile);
+    }
+  }
+};
+
+const findUnsupportedConfigs = localConfigs => {
```

Comment:
This is the main function, so we can skip the definition and call the implementation directly.

Reply:
You mean, just stick the contents of the function at the global level of the script?